esmero/format_strawberryfield
# ISSUE 18: Use general IIIF settings in Field Formatter default settings

## What's new?
Addresses [Issue 18](https://github.com/esmero/format_strawberryfield/issues/18)

- Added global default IIIF urls in `/config/install/iiif_settings.php`
- Added new abstract class `StrawberryBaseFormatter` that other all other FieldFormatters now extends. 
- The base class chooses the global IIIF urls by default but provides a checkbox UI for each formatter to select its own local urls.
- Added `/Tools/IiifFUrlValidator.php` which gets applied on the global IIIF Settings form and also each FieldFormatter's form for url setting (but passes over validating any localhost, 127.0.0.1, or 0.0.0.0 hosts).
- Refactored /Tools/IiifHelper.php to take a formatter's public and internal urls as arguments, makes for more efficient code in the formatter's `viewElements` method
- Changed `/schema/format_strawberryfield.schema.yml` to have a base formatter schema that then dynamically loads the formatter type and formatter-specific schema. The formatter-specific config is included under the `formatter_config` key of the base schema.

### How to Test:
- Test IIIF global defaults by reinstalling module
- Test Checkbox UI in the Manage Display tab of a Digital Object. Scroll down to open the formatter settings for a specific field. You will see the checkbox; toggle it and enter your own URLS.
- Then, Hitting "Update" will trigger the validation, so you can test that.
- If you save the overall form (after hitting Update in the individual formatter settings), and go back to the digital object, you should see the media being pulled from the new IIIF URL.
